tensorpack/dataflow/serialize.py

```diff
@@ -99,11 +99,9 @@ def save(df, path):
             path (str): output npz file.
         """
         buffer = []
-        size = _reset_df_and_get_size(df)
-        with get_tqdm(total=size) as pbar:
-            for dp in df.get_data():
-                buffer.append(dp)
-                pbar.update()
+        df.reset_state()
+        for dp in get_tqdm(df):
```

Comment:
This will fail if df has no size. `_reset_df_and_get_size(df)` was meant to provide a default size of 0.